source/plugins/go/src/extension/extension.go

```diff
@@ -98,33 +110,29 @@ func getExtensionSettings() (map[string]map[string]interface{}, error) {
 	return extensionSettings, nil
 }
 
-func getDataCollectionSettings() (map[string]string, error) {
-	dataCollectionSettings := make(map[string]string)
+func getDataCollectionSettingsInterface(extensionName string, extensionVersion string) (map[string]interface{}, error) {
+	dataCollectionSettings := make(map[string]interface{})
 
-	extensionSettings, err := getExtensionSettings()
+	extensionSettings, err := getExtensionSettings(extensionName, extensionVersion)
 	if err != nil {
 		return dataCollectionSettings, err
 	}
-	dataCollectionSettingsItr := extensionSettings["dataCollectionSettings"]
-	if len(dataCollectionSettingsItr) > 0 {
+
+	dataCollectionSettingsItr, ok := extensionSettings[EXTENSION_SETTINGS_DATA_COLLECTION_SETTINGS]
+	if ok && len(dataCollectionSettingsItr) > 0 {
 		for k, v := range dataCollectionSettingsItr {

```

Comment:
multiple places referring dataCollectionSettingsItr   might be able to extract as a function